crates/bevy_render/src/mesh/mesh.rs

```diff
@@ -369,6 +399,73 @@ impl Mesh {
 
         attributes_interleaved_buffer
     }
+
+    /// Duplicates the vertex attributes so that no vertices are shared.
+    ///
+    /// This can dramatically increase the vertex count, so make sure this is what you want.
+    /// Does nothing if no [Indices] are set.
+    pub fn duplicate_vertices(&mut self) {
+        fn duplicate<T: Copy>(values: &[T], indices: impl Iterator<Item = usize>) -> Vec<T> {
+            indices.map(|i| values[i]).collect()
+        }
+
+        assert!(
+            matches!(self.primitive_topology, PrimitiveTopology::TriangleList),
+            "can only duplicate vertices for `TriangleList`s"
+        );
+
+        let indices = match self.indices.take() {
+            Some(indices) => indices,
+            None => return,
+        };
+        for (_, attributes) in self.attributes.iter_mut() {
+            let indices = indices.iter();
+            match attributes {
+                VertexAttributeValues::Float(vec) => *vec = duplicate(&vec, indices),
+                VertexAttributeValues::Int(vec) => *vec = duplicate(&vec, indices),
+                VertexAttributeValues::Uint(vec) => *vec = duplicate(&vec, indices),
+                VertexAttributeValues::Float2(vec) => *vec = duplicate(&vec, indices),
+                VertexAttributeValues::Int2(vec) => *vec = duplicate(&vec, indices),
+                VertexAttributeValues::Uint2(vec) => *vec = duplicate(&vec, indices),
+                VertexAttributeValues::Float3(vec) => *vec = duplicate(&vec, indices),
+                VertexAttributeValues::Int3(vec) => *vec = duplicate(&vec, indices),
+                VertexAttributeValues::Uint3(vec) => *vec = duplicate(&vec, indices),
+                VertexAttributeValues::Int4(vec) => *vec = duplicate(&vec, indices),
+                VertexAttributeValues::Uint4(vec) => *vec = duplicate(&vec, indices),
+                VertexAttributeValues::Float4(vec) => *vec = duplicate(&vec, indices),
+                VertexAttributeValues::Uchar4Norm(vec) => *vec = duplicate(&vec, indices),
+            }
+        }
+    }
+
+    /// Calculates the [`Mesh::ATTRIBUTE_NORMAL`] of a mesh.
+    ///
+    /// Panics if [`Indices`] are set.
+    /// Consider calling [Mesh::duplicate_vertices] or export your mesh with normal attributes.
+    pub fn compute_flat_normals(&mut self) {
```

Comment:
Normals should also be normalized/unit length.

Reply:
Right, normals should be actually normal :smile: